### PR TITLE
remove _fork

### PR DESCRIPTION
**Pull request by zhreshold**
## Description ##
Fix shared Dataloader after removing _fork() in upstream mxnet. https://github.com/apache/incubator-mxnet/pull/12554

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- Interesting edge cases to note here

### CODE REVIEW

**Comment by szhengac** [comment]
The mxnet version in env needs to be updated.

**Comment by szha** [comment]
Let's hold onto this change so that we can release 0.4.1 w/ mxnet 1.3.0.

**Comment by mli** [comment]
Job PR-353/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-353/2/index.html

**Comment by mli** [comment]
Job PR-353/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-353/3/index.html

**Comment by mli** [comment]
Job PR-353/7 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-353/7/index.html

File: tests/unittest/train/test_dataloader.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import mxnet as mx
 from gluonnlp.data import FixedBucketSampler, ShardedDataLoader
 from mxnet import gluon
+from mxnet.test_utils import download
```
Comment (szha) [inline tests/unittest/train/test_dataloader.py]:
```suggestion
from mxnet.gluon.utils import download
```

File: tests/unittest/train/test_dataloader.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import mxnet as mx
```
Comment (szha) [inline tests/unittest/train/test_dataloader.py]:
```suggestion
import os
import mxnet as mx
```

File: tests/unittest/train/test_dataloader.py [hunk 3]
```diff
@@ -20,7 +21,15 @@ def test_sharded_data_loader():
                                        shuffle=False,
                                        num_shards=num_shards)
     for num_workers in [0, 1, 2, 3, 4]:
-        loader = ShardedDataLoader(dataset, batch_sampler=batch_sampler, num_workers=num_workers)
+        if num_workers == 2:
+            # test record file
+            url_format = 'https://apache-mxnet.s3-accelerate.amazonaws.com/gluon/dataset/{}'
+            filename = 'not_hotdog_validation-c0201740.rec'
+            download(url_format.format(filename))
```
Comment (szha) [inline tests/unittest/train/test_dataloader.py]:
```suggestion
            download(url_format.format(filename), path=os.path.join('tests', 'data', filename))
```

File: tests/unittest/train/test_dataloader.py [hunk 4]
```diff
@@ -20,7 +21,15 @@ def test_sharded_data_loader():
...
+            rec_dataset = gluon.data.RecordFileDataset(filename)
```
Comment (szha) [inline tests/unittest/train/test_dataloader.py]:
```suggestion
            rec_dataset = gluon.data.RecordFileDataset(os.path.join('tests', 'data', filename))
```

**Comment by mli** [comment]
Job PR-353/18 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-353/18/index.html

**Pull request merged**